src/main.rs

```diff
@@ -89,42 +101,80 @@ async fn main() -> Result<()> {
     engine.add_collector(Box::new(block_collector));
 
     let (batch_sender, batch_receiver) = channel(512);
+    let (priority_batch_sender, priority_batch_receiver) = channel(512);
     let (route_sender, route_receiver) = channel(512);
+    let (priority_route_sender, priority_route_receiver) = channel(512);
 
-    let uniswapx_collector = Box::new(UniswapXOrderCollector::new());
+    let uniswapx_collector = Box::new(UniswapXOrderCollector::new(chain_id, OrderType::Dutch));
     let uniswapx_collector =
         CollectorMap::new(uniswapx_collector, |e| Event::UniswapXOrder(Box::new(e)));
     engine.add_collector(Box::new(uniswapx_collector));
 
+    let priority_collector = Box::new(UniswapXOrderCollector::new(chain_id, OrderType::Priority));
+    let priority_collector =
+        CollectorMap::new(priority_collector, |e| Event::PriorityOrder(Box::new(e)));
+    engine.add_collector(Box::new(priority_collector));
+
     let uniswapx_route_collector =
-        Box::new(UniswapXRouteCollector::new(batch_receiver, route_sender));
+        Box::new(UniswapXRouteCollector::new(chain_id, batch_receiver, route_sender, args.executor_address.clone()));
     let uniswapx_route_collector = CollectorMap::new(uniswapx_route_collector, |e| {
         Event::UniswapXRoute(Box::new(e))
     });
     engine.add_collector(Box::new(uniswapx_route_collector));
 
+    let priority_route_collector = Box::new(UniswapXRouteCollector::new(
+        chain_id,
+        priority_batch_receiver,
+        priority_route_sender,
+        args.executor_address.clone(),
+    ));
+    let priority_route_collector = CollectorMap::new(priority_route_collector, |e| {
+        Event::UniswapXRoute(Box::new(e))
+    });
+    engine.add_collector(Box::new(priority_route_collector));
+
     let config = Config {
         bid_percentage: args.bid_percentage,
+        executor_address: args.executor_address,
     };
 
-    let strategy = UniswapXUniswapFill::new(
+    let uniswapx_strategy = UniswapXUniswapFill::new(
         Arc::new(provider.clone()),
-        config,
+        config.clone(),
         batch_sender,
         route_receiver,
     );
-    engine.add_strategy(Box::new(strategy));
+    engine.add_strategy(Box::new(uniswapx_strategy));
```

Comment:
maybe we want to run one or the other (v2 strat or prio strat), not both at once? I'd lean towards taking a `orderType` param

Reply:
yep great idea just added